src/Lean/Elab/Declaration.lean

```diff
@@ -353,15 +353,21 @@ def elabMutual : CommandElab := fun stx => do
     for builtin simprocs.
     -/
     let declNames ←
-       try
-         realizeGlobalConst ident
-       catch _ =>
-         let name := ident.getId.eraseMacroScopes
-         if (← Simp.isBuiltinSimproc name) then
-           pure [name]
-         else
-           throwUnknownConstant name
-    let declName ← ensureNonAmbiguous ident declNames
+      try some <$> realizeGlobalConst ident
```

Comment:
Don't we have `realizeGlobalConstWithInfo` or the like?

Reply:
The FIXME in this code is because we are using `realizeGlobalConst` even though sometimes it's not a valid declaration but a simproc instead. So some of the code from these functions is inlined here to ensure the error handling is done right.

I think it would be possible to use `realizeGlobalConstWithInfos` here and only do the ambiguity handling separately, although it has slightly different error behavior.